fnet/src/vespa/fnet/frt/supervisor.h

```diff
@@ -120,29 +108,37 @@ class FRT_Supervisor : public FNET_IServerAdapter,
     };
 
     // methods for performing rpc invocations
-    static void InvokeVoid(FNET_Connection *conn,
-                           FRT_RPCRequest *req);
-    static void InvokeAsync(SchedulerPtr scheduler,
-                            FNET_Connection *conn,
-                            FRT_RPCRequest *req,
-                            double timeout,
-                            FRT_IRequestWait *waiter);
-    static void InvokeSync(SchedulerPtr scheduler,
-                           FNET_Connection *conn,
-                           FRT_RPCRequest *req,
-                           double timeout);
+    static void InvokeVoid(FNET_Connection *conn, FRT_RPCRequest *req);
+    static void InvokeSync(SchedulerPtr scheduler, FNET_Connection *conn, FRT_RPCRequest *req, double timeout);
+    static void InvokeAsync(SchedulerPtr scheduler, FNET_Connection *conn, FRT_RPCRequest *req, double timeout, FRT_IRequestWait *waiter);
+
 
     // FNET ServerAdapter Interface
     bool InitAdminChannel(FNET_Channel *channel) override;
     bool InitChannel(FNET_Channel *channel, uint32_t pcode) override;
 
     // Packet Handling
     HP_RetCode HandlePacket(FNET_Packet *packet, FNET_Context context) override;
+};
 
-    // Methods for controlling transport object in standalone mode
-    bool Start();
-    void Main();
-    void ShutDown(bool waitFinished);
-    void WaitFinished();
+namespace fnet::frt {
+
+/**
+ * This is a simple class that makes it easy to test RPC.
+ * Normally you do not want use it in production code as it hides your possibilites and responsibilities.
+ */
+class StandaloneFRT {
+public:
+    StandaloneFRT();
+    explicit StandaloneFRT(std::shared_ptr<vespalib::CryptoEngine> crypto);
+    ~StandaloneFRT();
+    FRT_Supervisor & supervisor() { return *_supervisor; }
+    // TODO Remove this method as it is a relic from the ancient non-threaded world.
+    void wait_finished() const;
```

Comment:
I suggest just removing this one and letting the ancient tests/examples call it directly on the transport themselves

Reply:
Fixed